shark_turbine/kernel/wave/constraints.py

```diff
@@ -92,7 +103,27 @@ def subs_vector_shapes(self, index_map: dict[IndexSymbol, int]):
             if isinstance(vector_size, IndexExpr):
                 self.vector_shapes[vector_dim] = vector_size.subs(index_map)
 
-    def apply(self, mma_index: int) -> IndexSequence:
+    def compute_access_pattern_using_vector_shapes(
+        self,
+        dim: IndexSymbol,
+        workgroup_dim: int,
+        elements_per_thread: int | IndexSymbol,
+    ) -> IndexSequence:
+        if dim not in self.vector_shapes:
+            raise ValueError(f"No vector shape specified for dimension {dim}")
+        thread_id = self.get_thread_id_from_workgroup_dim(workgroup_dim)
+        return IndexSequence(thread_id * elements_per_thread, elements_per_thread, 1)
```

Comment:
Nice! now we have the quintessential elementwise index setting that can take `WorkgroupConstraint(M, BLOCK_M, 0) + element_per_thread` to `tid.x * elem_per_thread : tid.x * elem_per_thread + elem_per_thread ` indexing in M dim! :)

Reply:
You also need to specify vector_shape, but yes! :)

Reply:
But in this PR seems like we are only using vector_shapes to ensure that we are accessing the contiguous/fastest dimension right?

Reply:
When using vector_shapes we need the operators to define elements_per_thread which specifies the linear access pattern and elements_per_thread requires that the data being loaded be contiguous. So yes, use of vector_shapes implies contiguous accesses.